lib/options.go

```diff
@@ -175,6 +177,93 @@ func (ipnet *IPNet) UnmarshalText(b []byte) error {
 	return nil
 }
 
+// HostAddress stores information about IP and port
+// for a host.
+type HostAddress net.TCPAddr
+
+// NewHostAddress creates a pointer to a new address with an IP object.
+func NewHostAddress(ip net.IP, portString string) (*HostAddress, error) {
+	var port int
+	if portString != "" {
+		var err error
+		if port, err = strconv.Atoi(portString); err != nil {
+			return nil, err
+		}
+	}
+
+	return &HostAddress{
+		IP:   ip,
+		Port: port,
+	}, nil
+}
+
+// String converts a HostAddress into a string.
+func (h *HostAddress) String() string {
+	return (*net.TCPAddr)(h).String()
+}
+
+// MarshalText implements the encoding.TextMarshaler interface.
+// The encoding is the same as returned by String, with one exception:
+// When len(ip) is zero, it returns an empty slice.
+func (h *HostAddress) MarshalText() ([]byte, error) {
+	if h == nil || len(h.IP) == 0 {
+		return []byte(""), nil
+	}
+
+	if len(h.IP) != net.IPv4len && len(h.IP) != net.IPv6len {
+		return nil, &net.AddrError{Err: "invalid IP address", Addr: h.IP.String()}
+	}
+
+	return []byte(h.String()), nil
+}
+
+// UnmarshalText implements the encoding.TextUnmarshaler interface.
+// The IP address is expected in a form accepted by ParseIP.
+func (h *HostAddress) UnmarshalText(text []byte) error {
+	if len(text) == 0 {
+		return &net.ParseError{Type: "IP address", Text: "<nil>"}
+	}
+
+	ip, port, err := splitHostPort(text)
+	if err != nil {
+		return err
+	}
+
+	nh, err := NewHostAddress(ip, port)
+	if err != nil {
+		return err
+	}
+
+	*h = *nh
+	return nil
+}
+
+func splitHostPort(text []byte) (net.IP, string, error) {
+	host := string(text)
+	var port string
+
+	if isHostPort(text) {
+		var err error
+		host, port, err = net.SplitHostPort(host)
+		if err != nil {
+			return nil, "", err
+		}
+	}
+
+	ip := net.ParseIP(host)
+	if ip == nil {
+		return nil, "", &net.ParseError{Type: "IP address", Text: host}
+	}
+
```

Comment:
are you here trying to figure out if there is a port ? so that `net.SplitHostPort` doesn't return you an error?  Because IMO you can just call it - and if there is no error - there is a port, if there is an error - there is no port :D 
As far as I can see this code currently doesn't handle `something.com:8413` for example correctly ?

Reply:
I've removed the double check.

>  As far as I can see this code currently doesn't handle something.com:8413 for example correctly ?

I don't understand this commend. As far as I can tell, the current code expects the option to be a valid IP address, and `something.com` won't ever work as IP address.